### PR TITLE
fix: failed loading message

### LINKED ISSUE #1602: TypeError: Cannot read property 'failed' of undefined
When polymer project is down smokehouse fails with the error below instead of correctly falling into the branch of "Page failed to load"

```
Runtime error encountered: TypeError: Cannot read property 'failed' of undefined
    at pass.then.then.networkRecords (/Users/phulce/Code/Chrome/lighthouse/lighthouse-core/gather/gather-runner.js:211:38)
TypeError: Cannot read property 'failed' of undefined
    at pass.then.then.networkRecords (/Users/phulce/Code/Chrome/lighthouse/lighthouse-core/gather/gather-runner.js:211:38)
```

### PR DESCRIPTION
**Pull request by patrickhulce**
fixes #1602

### CODE REVIEW

File: lighthouse-core/gather/gather-runner.js [hunk 1]
```diff
@@ -208,9 +208,12 @@ class GatherRunner {
       return driver.endNetworkCollect();
     }).then(networkRecords => {
       const mainRecord = networkRecords.find(record => record.url === options.url);
-      if (driver.online && mainRecord.failed) {
-        log.error('GatherRunner', mainRecord.localizedFailDescription);
-        const error = new Error(`Unable to load the page: ${mainRecord.localizedFailDescription}`);
+      if (driver.online && (!mainRecord || mainRecord.failed)) {
+        const failDescription = mainRecord ?
```
Comment (ebidel) [inline lighthouse-core/gather/gather-runner.js]: nit: this can probably fit on one line.

**Comment by ebidel** [comment]
Ironically, tests are failing b/c of shop.

File: lighthouse-core/gather/gather-runner.js (hunk 1 above)
Comment (patrickhulce) [inline lighthouse-core/gather/gather-runner.js]: not quite, should I make the name shorter?

**Comment by patrickhulce** [comment]
> Ironically, tests are failing b/c of shop.

But isn't the error message you get so much friendlier? 😄

File: lighthouse-core/gather/gather-runner.js (hunk 1 above)
Comment (ebidel) [inline lighthouse-core/gather/gather-runner.js]: fitting would be nice, but not a big deal.

**Comment by patrickhulce** [comment]
will do 👍

**Comment by patrickhulce** [comment]
PTAL

File: lighthouse-core/gather/gather-runner.js [hunk 2]
```diff
@@ -125,6 +125,22 @@ class GatherRunner {
   }
 
   /**
+   * @param {string} url
+   * @param {{online: boolean}} driver
+   * @param {!Array<WebInspector.NetworkRequest} networkRecords
```
Comment (brendankenny) [inline lighthouse-core/gather/gather-runner.js]: need a closing `>`

File: lighthouse-core/gather/gather-runner.js [hunk 3]
```diff
@@ -125,6 +125,22 @@ class GatherRunner {
   }
 
   /**
+   * @param {string} url
```
Comment (brendankenny) [inline lighthouse-core/gather/gather-runner.js]: single line about what it's doing?

File: lighthouse-core/test/gather/gather-runner-test.js [hunk 4]
```diff
@@ -511,6 +511,30 @@ describe('GatherRunner', function() {
...
+    });
+
+    it('throws when page times out', () => {
+      const url = 'http://the-page.com';
+      const records = [];
+      assert.throws(() => {
+        GatherRunner.assertPageLoaded(url, {online: true}, records);
+      }, /Unable.*timeout/);
+    });
+  });
```
Comment (brendankenny) [inline lighthouse-core/test/gather/gather-runner-test.js]: worth moving `rejects when domain name can\'t be resolved` and `resolves when domain name can\'t be resolved but is offline` tests into this `describe` block? They do test the whole pipeline, not just the function, but they're more or less just unit tests of the function...

Comment (patrickhulce) [inline lighthouse-core/test/gather/gather-runner-test.js]: I've noticed we have some differing philosophies when it comes to `describe` blocks and test organization in general, haha, so I'm opposed but will defer to you if you want 'em moved :)

File: lighthouse-core/gather/gather-runner.js (hunk 2 above)
Comment (patrickhulce) [inline lighthouse-core/gather/gather-runner.js]: done

File: lighthouse-core/gather/gather-runner.js (hunk 3 above)
Comment (patrickhulce) [inline lighthouse-core/gather/gather-runner.js]: done

File: lighthouse-core/test/gather/gather-runner-test.js (hunk 4 above)
Comment (brendankenny) [inline lighthouse-core/test/gather/gather-runner-test.js]: I have no coherent philosophy when it comes to organization of tests into sub-blocks so keeping as-is is good :)

**Comment by brendankenny** [comment]
☁️↔️✂️↔️💻 ⚠️️

**Pull request merged**